Dev/issue merge xliff into weblate

sbreker opened the pull request:
### Merge AtoM XLIFF to Weblate

Add a new option to calls to import_from_atom to choose whether XLIFF
files are copied over destination XLIFF files, or source strings are
merged into the target file. Merging is used when updating AtoM strings
into Weblate work-in-progress translation strings.

### Remove 'translated' XLIFF attribute

Remove all references to the 'translated' attribute on the
tag. This attribute is not valid XLIFF and is ignored by Weblate.

### Removed import compare

Removed comparison of AtoM strings to Weblate strings in export_to_atom
because the comparison is unnecessary as the files will never be the
same.

### Merge files into AtoM

Merge XLIFF into AtoM's .tx folder instead of overwriting these files
with the Weblate versions.

When merging new translations from Weblate into AtoM, the scripts were
overwriting AtoM's .tx files before the phing scripts ran to import
these strings into AtoM's modules/fixtures etc. This change adds the
ability to merge the Weblate XLIFF into the AtoM XLIFF. This preserves
source strings that are not flagged as "Approved" or have been
accidentally changed in Weblate.

This corrects situations where strings were being removed when the source
string was altered in Weblate.

sbreker committed:
Merge AtoM XLIFF to Weblate

Add a new option to calls to import_from_atom to choose whether XLIFF
files are copied over destination XLIFF files, or source strings are
merged into the target file.

Merging is used when updating AtoM strings into Weblate work-in-progress
translation strings.

sbreker committed:
Do not always set to Approved

sbreker committed:
Remove 'translated' XLIFF attribute

Remove all references to the 'translated' attribute on the <trans-unit>
tag. This attribute is not valid XLIFF and is ignored by Weblate.

Weblate now manages this using the 'state' attribute on the <target>
tag. Managing the 'state' will be transparent to AtoM. If the 'state'
is not present, Weblate uses the presence of target string to determine
the current state.

sbreker committed:
Removed import compare

Removed comparison of AtoM strings to Weblate strings in export_to_atom
because the comparison is unnecessary as the files will never be the
same.

sbreker committed:
Merge files into AtoM

Merge XLIFF into AtoM's .tx folder instead of overwriting these files
with the Weblate versions.

When merging new translations from Weblate into AtoM, the scripts were
overwriting AtoM's .tx files before the phing scripts ran to import
these strings into AtoM's modules/fixtures etc. This change adds the
ability to merge the Weblate XLIFF into the AtoM XLIFF. This preserves
source strings that are not flagged as "Approved" or have been
accidentally changed in Weblate.

This corrects situations where strings were being removed when the source
string was altered in Weblate.

djjuhasz commented on README.md:
```diff
@@ -112,8 +112,8 @@ Example import:
 In addition to copying the XLIFF files it will also process them so they're
-suitable for Weblate (among other things, adding "approved" and "translated"
-attributes to each XLIFF translation unit that's already been translated).
+suitable for Weblate (among other things, adding "approved" attribute to 
```
Nitpick: "adding *an* "approved" attribute to"

djjuhasz reviewed (approved):
Looks good @sbreker! I just flagged one minor grammar issue. :)

sbreker committed:
Fix merge to AtoM error

Fix error when merging Weblate XLIFF to AtoM that occurs when the
Weblate file contains zero strings. This can happen when all strings are
'unapproved' in Weblate. In this case use AtoM's XLIFF and skip merging.
